From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject: s390/kernel: use stnsm 255 instead of stosm 0
Patch-mainline: v3.19-rc5
Git-commit: 81fc77fbfc1553aee35e33af89959be1d8c81db2
References: bnc#926797, LTC#KRN1514

Summary:     s390: z13 base performance
Description: Provide the backport of a minimum set of upstream patches
             to optimize the base performance of the IBM z13 machine.

Upstream-Description:

             s390/kernel: use stnsm 255 instead of stosm 0

             On some models, stnsm 255 might be slightly faster than stosm 0.

             Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>
             Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/include/asm/irqflags.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/s390/include/asm/irqflags.h
+++ b/arch/s390/include/asm/irqflags.h
@@ -36,7 +36,7 @@ static inline void __arch_local_irq_ssm(
 
 static inline unsigned long arch_local_save_flags(void)
 {
-	return __arch_local_irq_stosm(0x00);
+	return __arch_local_irq_stnsm(0xff);
 }
 
 static inline unsigned long arch_local_irq_save(void)
